### PR TITLE
doubly_linked_list.c Add Doubly Linked List C Implementation

### LINKED ISSUE #31: [DataStructure] Doubly Linked List [C]

### PR DESCRIPTION
**Pull request by arc9693**
This adds Algorithm for implementation of Doubly Linked List Data Structure.
Each node is represented by struct data type that contains integer variable data
for storing integer data,one pointer that points to next node and another that
pointer that points to previous node in the linked list.

Closes NITSkmOS#31

Fixes #31 

**By submitting this pull request I confirm I've read and complied with the
below declarations.**

- [x] I have read the [Contribution guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [commit guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#write-good-commit-messages) for this project.
- [x] My code follows the [standard code structure](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#samples-code).
- [x] This pull request has a descriptive title. For example, `{Tag}: Add {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.
- [x] This pull request shall only be reviewed and merged once all the checks passes. No maintainer or supporter shall be obliged to review it before this condition is met.
- [x] I have mentioned the issue number correctly (with hyperlink) in this pull request description.

### CODE REVIEW

**Comment by arc9693** [comment]
@sangamcse please review.

**Comment by sangamcse** [comment]
Please add into README Table also.

And please use `git commit --amend` instead of `git commit` while committing. :)

**Comment by sangamcse** [comment]
@arc9693, your checks are failing

File: README.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ This repository contains examples of various algorithms written on different pro
 | [Stack](https://en.wikipedia.org/wiki/Stack_(abstract_data_type))                               |                                       |                                       | [:octocat:](stack/Java)               | [:octocat:](stack/Python)             |
 | [Linear Linked List](https://en.wikipedia.org/wiki/Linked_list)                                 | [:octocat:](linked_list/C)            |                                       |                                       |                                       |
 | [AVL Tree](https://en.wikipedia.org/wiki/AVL_tree)                                              | [:octocat:](avl_tree/C)               |                                       | [:octocat:](avl_tree/Java)            |                                       |
-
+| [Doubly Linked List](https://en.wikipedia.org/wiki/Doubly_linked_list) |[:octocat:](doubly_linked_list/c) |
```
Comment (sangamcse) [inline README.md]: @arc9693, there is more blank column we need to add. Please do that too.

File: README.md [hunk 2]
```diff
@@ -49,7 +49,6 @@ This repository contains examples of various algorithms written on different pro
 2. Clone the forked repository to local system.
 3. Create your feature branch: `git checkout -b my-new-feature`
 4. Commit your changes: `git commit -am 'Add some feature'`.
-
```
Comment (sangamcse) [inline README.md]: And we need this line break too

File: doubly_linked_list/c/doubly_linked_list.c [hunk 3]
```diff
@@ -0,0 +1,91 @@
+#include <stdio.h>
```
Comment (sangamcse) [inline doubly_linked_list/c/doubly_linked_list.c]: We require code comments for the code

**Pull request closed without merging**